tcp-source/src/main/java/org/springframework/cloud/stream/module/tcp/source/TcpSourceProperties.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.module.tcp.source;
+
+import javax.validation.constraints.NotNull;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.stream.module.tcp.AbstractTcpConnectionFactoryProperties;
+
+/**
+ * Properties for the TCP Source.
+ *
+ * @author Gary Russell
+ *
+ */
+@ConfigurationProperties
+public class TcpSourceProperties extends AbstractTcpConnectionFactoryProperties {
+
+	/**
+	 * The decoder to use when receiving messages. CRLF, LF, NULL, STXETX, L1, L2, L4, RAW.
```

Comment:
It's not in the json, but code completion knows how to discover enum values. see https://github.com/spring-cloud/spring-cloud-dataflow/blob/master/spring-cloud-dataflow-completion/src/main/java/org/springframework/cloud/dataflow/completion/EnumValueHintProvider.java